dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
-#     self-discovery to set this. You can combine this with values of DOCKER_HOST to 
-#     change communications from socket to TCP.
+#
+# Affects how the Codenvy server connects to the Docker Daemon, how the Che server and
+# workspaces establish connections to one another, and also how remote browser clients
+# should discover and connect to workspaces.
+#
+# Codenvy goes through a progression algorithm to establish the protocol, IP address and
+# port to establish communications with it is booting or starting a workspace.
+#
+# Browser --> Codenvy Server
+#    1. Default is 'http://localhost:${SERVER_PORT}/wsmaster/api'.
+#    2. Else use the value of CODENVY_API
+#
+# Codenvy Server --> Docker Daemon Progression:
+#    1. Use the value of CODENVY_DOCKER_DAEMON__URL
+#    2. Else, use the value of DOCKER_HOST system variable
+#    3. Else, use Unix socket over unix:///var/run/docker.sock
+#
+# Codenvy Server --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
+#        1. Use the value of CODENVY_DOCKER_IP
+#        2. Else, if server connects over Unix socket, then use localhost
+#        3. Else, use DOCKER_HOST
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'docker-local':
+#        1. Use the address of the workspace container within the docker network 
+#           and exposed ports
+#        2. If address is missing, if server connects over Unix socket, then use 
+#           localhost and exposed ports
+#        3. Else, use DOCKER_HOST and published ports
+#
+# Browser --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else if set use the value of CODENVY_DOCKER_IP
+#        3. Else, if server connects over Unix socket, then use localhost
+#        4. Else, use DOCKER_HOST
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'docker-local':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else use the address of the workspace container within the docker network, 
+#           if it is set
+#        3. If address is missing, if server connects over Unix socket, then use 
+#           localhost
+#        4. Else, use DOCKER_HOST
+#
+# Workspace Agent --> Codenvy Server
+#    1. Default is 'http://codenvy-host:${SERVER_PORT}/wsmaster/api', where 
+#       'codenvy-host' is IP of server.
+#    2. Else, use value of CODENVY_WORKSPACE_CHE__SERVER__ENDPOINT
+#    3. Else, if 'docker0' interface is unreachable, then 'codenvy-host' replaced 
+#       with 172.17.42.1 or 192.168.99.1
+#    4. Else, print connection exception
+
+# Codenvy Server API Endpoint
+#     The location of the API end point where dashboard and IDE clients will look for
+#     interacting with the Codenvy server, which we also call the workspace master.
+#CODENVY_API=http://localhost:${SERVER_PORT}/wsmaster/api
```

Comment:
You tell me.  These are just copies over from Che.  I am hoping you tell me which properties are used by Codenvy and which ones are not.  There is a CHE_API for Che.  It's very possible it does not exist, but need engineers to tell me.